test/e2e/operator_test.go

```diff
@@ -3126,6 +3126,139 @@ func TestAWSELBConnectionIdleTimeout(t *testing.T) {
+	}
+	if err := kclient.Create(context.Background(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	t.Cleanup(func() { assertIngressControllerDeleted(t, kclient, ic) })
+
+	operatorImage, err := getIngressOperatorDeploymentImage(t, kclient, 1*time.Minute)
+	if err != nil {
+		t.Fatal("failed to determine ingress operator deployment's image: ", err)
+	}
+
+	iptablesImage, err := getIptablesImage(t, kclient, 1*time.Minute)
+	if err != nil {
+		t.Fatal("failed to determine image with iptables tool: ", err)
+	}
+
+	// Create a pod with an HTTP application that delays the connection and sends echo responses.
+	httpdPod := buildDelayConnectHTTPPod("connect-timeout-http", operatorcontroller.DefaultOperandNamespace, iptablesImage, operatorImage)
+	if err := kclient.Create(context.Background(), httpdPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", httpdPod.Namespace, httpdPod.Name, err)
+	}
+	t.Cleanup(func() {
+		if err := kclient.Delete(context.Background(), httpdPod); err != nil {
+			t.Fatalf("failed to delete pod %s/%s: %v", httpdPod.Namespace, httpdPod.Name, err)
+		}
+	})
+
+	if err := waitForPodReady(t, kclient, httpdPod, 2*time.Minute); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	httpdService := buildEchoService(httpdPod.Name, httpdPod.Namespace, httpdPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.Background(), httpdService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", httpdService.Namespace, httpdService.Name, err)
+	}
+	t.Cleanup(func() {
+		if err := kclient.Delete(context.Background(), httpdService); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", httpdService.Namespace, httpdService.Name, err)
+		}
+	})
+
+	route := buildRoute(httpdPod.Name, httpdPod.Namespace, httpdService.Name)
+	route.Spec.Host = fmt.Sprintf("%s-%s.%s", route.Name, route.Namespace, ic.Spec.Domain)
+	if err := kclient.Create(context.Background(), route); err != nil {
+		t.Fatalf("failed to create route %s/%s: %v", route.Namespace, route.Name, err)
+	}
+	t.Cleanup(func() {
+		if err := kclient.Delete(context.Background(), route); err != nil {
+			t.Fatalf("failed to delete route %s/%s: %v", route.Namespace, route.Name, err)
+		}
+	})
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, icName, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Get the ELB's hostname via the wildcard DNS record
+	wildcardRecordName := controller.WildcardDNSRecordName(ic)
+	wildcardRecord := &iov1.DNSRecord{}
+	if err := kclient.Get(context.Background(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+	elbHostname := wildcardRecord.Spec.Targets[0]
```

Comment:
*nit* You imply ELBs here, but this test is not restricted to just AWS. Looks like it works fine on GCP and Azure, but shall we use a more generic term, like just lb or load balancer? I don't think they are called ELBs on GCP and azure.

Also, your comments mention ELBs as well.

Reply:
Done.